File: analyze_protein_trajectory/HP36/traj_A/calc_Eu_plus_xmu/calc_Eu_plus_xmu.py
import sys
import pandas as pd

"""
/homes/eta/users/aiteam/HP36_tutorial/tools/calc_Eu_plus_xmu/

input  : tenergy.dat
         xmu_vs_time.dat

output : Eu_vs_time.dat
         Eu_plus_xmu_vs_time.dat
"""

# load data files
df_e = pd.read_csv('tenergy.dat', header=None, skiprows=1, delimiter='\s+')
df_e.columns = ['time','Eu','e_bond','e_angle','e_dihedral','e_14LJ','e_14elec','e_LJ','e_elec']
# print(df_e.head())

df_xmu = pd.read_csv('xmu_vs_time.dat', header=None, delimiter='\s+')
df_xmu.columns = ['time','xmu']
# print(df_xmu.head())

num0 = df_e.shape[0]
num1 = df_xmu.shape[0]


# check the number of data in energy.dat and xmu_vs_time.dat
if num0 != num1:
    print('error: inconsistency in the number of data')
    sys.exit()

print('total number of data : %s' % num0)

# manipulation
time_interval = 10  # ps

file_Eu = open('Eu_vs_time.dat', 'w')
file_total = open('Eu_plus_xmu_vs_time.dat', 'w')
file_Eu_250 = open('Eu_vs_time_every_250ps.dat', 'w')
file_total_250 = open('Eu_plus_xmu_vs_time_every_250ps.dat', 'w')

for i in range(num0):
    time0 = int(df_e.iloc[i].time * time_interval)
    time1 = int(df_xmu.iloc[i].time)

    Eu = df_e.iloc[i].Eu
    xmu = df_xmu.iloc[i].xmu

    if time0 != time1:
        print('error:  inconsistency in times.',time0,time1)
        sys.exit()


    print('{:>5d}\t{:>15.8f}'.format(time0, Eu), file=file_Eu)
    print('{:>5d}\t{:>15.8f}'.format(time0, Eu+xmu), file=file_total)

    if time0 % 250 == 0:
        print('{:>5d}\t{:>15.8f}'.format(time0, Eu), file=file_Eu_250)
        print('{:>5d}\t{:>15.8f}'.format(time0, Eu+xmu), file=file_total_250)


file_Eu.close()
file_total.close()
file_Eu_250.close()
file_total_250.close()

# =============================================================================

File: analyze_protein_trajectory/py_tools_multi_chains/calc_Eu_plus_xmu/1000_ps_interval/calc_Eu_plus_xmu.f
c
c       input  : energy.dat
c                xmu_vs_time.dat
c
c       output : Eu_vs_time.dat
c                Eu_plus_xmu_vs_time.dat
c

      program calc_Eu_plus_xmu
      implicit none

      real*8  time0, Eu
      real*8  time1, xmu

      real*8  dum1, dum2, dum3, dum4, dum5, dum6, dum7
      integer i, n
      integer n1, n2

      integer mod_diff

      character*120 dummy

c
c     check the number of data in energy.dat and xmu_vs_time.dat
c

      n1 = 0
      n2 = 0

      open(10,file='energy.dat',status='old',err=90)
      read(10,'(a)',err=90)     dummy ! for comment line in energy.dat
  100 read(10,*,end=190,err=90) time0, Eu, 
     .                          dum1,dum2,dum3,dum4,dum5,dum6,dum7
      n1 = n1+1
      goto 100

  190 continue
      close(10,err=90)

      open(11,file='xmu_vs_time.dat',status='old',err=90)
  200 read(11,*,end=290,err=90) time1, xmu
      n2 = n2+1
      goto 200

  290 continue
      close(11,err=90)

      if(n1.ne.n2) then
        write(6,*) 'error: inconsistency in the number of data'
        goto 99
      endif

      n = n1

      write(6,*) 'total number of data : ',n

c
c     manipulation
c

      open(10,file='energy.dat',status='old',err=90)
      open(11,file='xmu_vs_time.dat',status='old',err=90)

      open(20,file='Eu_vs_time.dat',err=90)
      open(21,file='Eu_plus_xmu_vs_time.dat',err=90)

      read(10,'(a)',err=90)  dummy ! for comment line in energy.dat

      do i=1,n

        read(10,*,err=90) time0, Eu
        read(11,*,err=90) time1, xmu

        time0 = time0*1000.d0

c       if(abs(time0-time1).gt.1.0d-3) then
c         write(6,*) 'error:  inconsistency in times',time0,time1
c         goto 99
c       endif

        mod_diff = mod(int(time0-time1),100000) ! 100000 means 100 ns

c       write(6,*) time0, time1, mod_diff

        if(mod_diff.ne.0) then
          write(6,*) 'error:  inconsistency in times',time0,time1
          goto 99
        endif

        write(20,900,err=90) time1,Eu
        write(21,900,err=90) time1,Eu+xmu

      enddo

 900  format(f16.8,1x,f16.8)

      close(10,err=90)
      close(11,err=90)

      close(20,err=90)
      close(21,err=90)

c..........
   99 stop
   90 write(6,*) 'MAIN:  I/O error'
      goto 99
c
      end

